### DISCUSSION TITLE
Error loading config

### DISCUSSION THREAD

**Discussion by fabiopittol** [jetimob/juno-sdk-php-laravel#20] [Q&A]
Hi, 
I'm trying to use the package on a project here for the first time (so excuse any dumbness). I followed the installation steps correctly. Didn't quite get the `use Juno;` import but used `use Jetimob\Juno\Juno;` inside the desired controller instead, then I ran:

`$response = (new Juno)->request('', '');`

I know the attributes will fail as empty. But before that, I get following error: 

> ErrorException
> Undefined index: request_max_attempts

Also, using `Juno::request` as suggested on the docs gives me an 'non static method' error, that's why I changed to that.

I believe I must be importing it wrong, aren't I?

**Comment by alanwgt** [comment]
Hello!

There is no dumb question, please feel free to ask anything :)

The `use Juno` is just an alias to `use Jetimob\Juno\Facades\Juno`. You need to call the methods on the facade so that the Juno instance is correctly instantiated using the config file in your project, and the methods are executed on the instance through the facade. *You must import the facade, not the Juno class*.

[Take this file as an example](https://github.com/jetimob/juno-sdk-php-laravel/blob/master/src/tests/Feature/AccountTestCase.php), you can see that we import `Juno` with `use Jetimob\Juno\Facades\Juno;`, and [here](https://github.com/jetimob/juno-sdk-php-laravel/blob/master/src/tests/Feature/AccountTestCase.php#L69) it works as expected.

Let me know if this fixes your issue!

**Reply by fabiopittol** [reply]
Hey Alan, yes it worked! Thanks a lot!

Imported as
```
use Jetimob\Juno\Facades\Juno as Juno;
use Jetimob\Juno\Lib\Http\Document\DocumentListRequest;
```
then ran
```
        $response = Juno::request(DocumentListRequest::class, '');
        return $response;
```
Returned `{ "timestamp": 1628020603, "statusCode": 401, "error": "Unauthorized", "details": [ {} ], "path": "\/documents" }`
Which despite being an error, at least it seems to have connected.

One last question to double check, tho: Can I always leave the `$resourceToken` empty for the request? That way it will be checked against the cache or generate a new one with the credentials?

**Reply by alanwgt** [reply]
I'm glad to hear that it worked!

For most of the API calls, the `$resourceToken` **cannot** be empty. This is because this token identifies the account in Juno's API.

From the [docs](https://dev.juno.com.br/api/v2#section/Servidor-de-Recursos):

> Muitos dos recursos também necessitam de um token de recurso, X-Resource-Token que identifica a conta digital que deverá ser utilizada durante a execução de uma operação.
>
> Cada conta digital tem o seu próprio token de recurso. Contas digitais criadas via API incluem o token de recurso na resposta da requisição.
>
> Para obter o token de recurso de uma conta digital já existente ou para redefinir o token de recurso, o cliente precisa acessar o painel do cliente Juno e realizar esta operação na aba Integração, opção Token Privado.

I hope that this clears any doubt about the `$resourceToken`. If not, feel free to ask :)

Cheers!

**Reply by fabiopittol** [reply]
So, it can be the same as the `JUNO_PRIVATE_TOKEN` on the `.env` file, which holds the token generated via Juno's dashboard with the _**"Gerar token privado"**_ option?

They talk about so many tokens, I got confused lol

**Reply by alanwgt** [reply]
Exactly!

You can safely interchange *private token* with *resource token*, they are the same thing.

**Reply by fabiopittol** [reply]
Great! Thanks a lot, Alan!

**Reply by alanwgt** [reply]
Anytime!
